src/Implementations.py

```diff
@@ -0,0 +1,54 @@
+# ║  ╔═╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩═╗  ║
+# ╠══╣                                                                                                            ╠══╣
+# ║  ║    IMPLEMENTATIONS                         CREATED: 2024-09-24          https://github.com/jacobleazott    ║  ║
+# ║══║                                                                                                            ║══║
+# ║  ╚═╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦══════╦═╝  ║
+# ╚════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩══════╩════╝
+# ═══════════════════════════════════════════════════ DESCRIPTION ════════════════════════════════════════════════════
+# 
+# ════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+from Spotify_Features import SpotifyFeatures
+import General_Spotify_Helpers as gsh
+from Shuffle_Styles import ShuffleType
+
+def main():
+    features = SpotifyFeatures(log_file_name="Playback.log")
+    track_id, shuffle_enabled, playlist_id = features.get_playback_state()
+
+    # ════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+    # TRACK MACROS ═══════════════════════════════════════════════════════════════════════════════════════════════════
+    # ════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+    match track_id:
+        case gsh.SHUFFLE_MACRO_ID:
+            shuffle_type = ShuffleType.WEIGHTED if shuffle_enabled else ShuffleType.RANDOM
+            features.shuffle_playlist(playlist_id, shuffle_type)
+            
+        case gsh.GEN_ARTIST_MACRO_ID:
+            features.skip_track()
+            features.generate_artist_playlist_from_playlist(playlist_id)
+            
+        case gsh.DISTRIBUTE_TRACKS_MACRO_ID:
+            features.skip_track()
+            features.distribute_tracks_to_collections(playlist_id)
+            
+        case gsh.ORGANIZE_PLAYLIST_MACRO_ID:
+            features.skip_track()
+            features.organize_playlist_by_date(playlist_id)
+            
+        case _:
+            features.log_playback_to_db(track_id)
+        
+    # ════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+    # DATE TRIGGERS ══════════════════════════════════════════════════════════════════════════════════════════════════
+    # ════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+        # Check Date/ Time
+        #   Trigger Weekly Report
+        #   Trigger Daily Backup
+        #   Trigger Monthly Release
+            
+
+if __name__ == "__main__":
```

Comment:
Checking that I understand what's going on here - is this if re-calling main to make it a loop? I was expecting this to have a forever loop

Reply:
Nvm, I looked at the cron jobs and `Proj.sh`, so it makes sense no loop here. I'm still a little confused by this syntax, but what else is new

Reply:
if __name___... basically means that this file was being called directly ie. 'python Monthly_Release.py' the benefit of this is it means we can have functionality only when specifically called, but also when it is being inherited by other files that main() would not trigger since __name__ would not be __main__.